vue-js/src/components/task1/List.vue

```diff
@@ -0,0 +1,60 @@
+  <template>
+  <div id="list">
+	<h3>{{headline}}</h3>
+	<ul>
+      <list-item v-for="(item, index) in ToDoList"
+                 v-bind:item='item'
+                 v-bind:key=index
+                 v-on:delete-item="deleteThisItem(index)">
+      </list-item>
+      
+	</ul>
+  <div id="add">
+        <input id="add-item-text" />
+        <button v-on:click="addText">Add</button>
+  </div>
+  </div>
+</template>
+
+<script>
+import ListItem from "./ListItem" 
+
+export default {
+  name: "List",
+  components: {
+    "list-item" : ListItem
+  },
+  data: function() {
+	return {
+      headline: "ToDo-List",
+      ToDoList: [
+        {text: 'Clean room'},
+        {text: 'Finish task1'},
+        { text: 'Add task2'}
+      ]
+    }
+  },
+  methods: {
+    deleteThisItem: function(index) {
+      this.ToDoList.splice(index, 1);
+    },
+    addText: function(){
+      this.ToDoList.push({
+        text:  document.getElementById("add-item-text").value
```

Comment:
Try to avoid globals like `document` or `window`. It's not relevant yet, but we will cover server-side rendering during the lecture and those globals are not available outside the browser environment.

Reply:
Apart from that, using direct DOM manipulations is not quite the VueJS way, which utilizes a virtual DOM.